packages/@vuepress/core/lib/node/App.js

```diff
@@ -348,7 +348,12 @@ module.exports = class App {
       computed: new this.ClientComputedMixinConstructor(),
       enhancers: this.pluginAPI.getOption('extendPageData').items
     })
-    this.pages.push(page)
+    const index = this.pages.findIndex(({ path }) => path === page.path)
+    if (index >= 0) {
+      this.pages.splice(index, 1, page)
```

Comment:
It would be better to emit a warning with previous and actual active route here.

Reply:
@ulivz Updated.